src/middlewares/asyncWrapper.js

```diff
@@ -0,0 +1,23 @@
+
+/**
+  * A wrapper for express js controller for error handling
+  *
+  * @function
+  *
+  * @param {Function} wrappedFunction - the main controller
+  *
+  * @return {Function} - callback that execute the controller
+  */
+export default (wrappedFunction) => {
+  const currentFunction = wrappedFunction;
```

Comment:
Good implementation. What's the point of reassigning it though?

Reply:
None actually. just a preference.
I will remove that